tests/conftest.py

```diff
@@ -29,42 +74,131 @@ def complex_data():
+
+
+@pytest.fixture(scope="function")
+def complex_data_split_with_categorical(complex_data_split):
+    X_train, X_test, y_train, y_test = complex_data_split
+    X_train["f1_categorical"] = X_train["f1_categorical"].astype(str).astype("category")
+    X_test["f1_categorical"] = X_test["f1_categorical"].astype(str).astype("category")
+
     return X_train, X_test, y_train, y_test
 
 
 @pytest.fixture(scope="function")
-def complex_lightgbm():
+def complex_lightgbm(random_state_42):
+    """This fixture allows to reuse the import of the LGBMClassifier class across different tests."""
+    model = LGBMClassifier(max_depth=5, num_leaves=11, class_weight="balanced", random_state=random_state_42)
+    return model
+
+
+@pytest.fixture(scope="function")
+def complex_fitted_lightgbm(complex_data_split_with_categorical, complex_lightgbm):
     """
     Fixture.
     """
-    import lightgbm
+    X_train, _, y_train, _ = complex_data_split_with_categorical
 
-    return lightgbm.LGBMClassifier(max_depth=5, num_leaves=11, class_weight="balanced", random_state=42)
+    return complex_lightgbm.fit(X_train, y_train)
 
 
 @pytest.fixture(scope="function")
-def complex_fitted_lightgbm(complex_data_split, complex_lightgbm):
+def catboost_classifier(random_state):
+    """This fixture allows to reuse the import of the CatboostClassifier class across different tests."""
+    model = CatBoostClassifier(random_seed=random_state)
+    return model
+
+
+@pytest.fixture(scope="function")
+def decision_tree_classifier(random_state):
+    """This fixture allows to reuse the import of the DecisionTreeClassifier class across different tests."""
+    model = DecisionTreeClassifier(max_depth=1, random_state=random_state)
+    return model
+
+
+@pytest.fixture(scope="function")
+def randomized_search_decision_tree_classifier(decision_tree_classifier, random_state):
+    """This fixture allows to reuse the import of the DecisionTreeClassifier in combination with a new CV class across different tests."""
+    param_grid = {"criterion": ["gini"], "min_samples_split": [1, 2]}
+    cv = RandomizedSearchCV(decision_tree_classifier, param_grid, cv=2, n_iter=2, random_state=random_state)
+    return cv
+
+
+@pytest.fixture(scope="function")
+def logistic_regression(random_state):
+    """This fixture allows to reuse the import of the DecisionTreeClassifier class across different tests."""
+    model = LogisticRegression(random_state=random_state)
+    return model
+
+
+@pytest.fixture(scope="function")
+def X_train():
     """
     Fixture.
     """
-    X_train, _, y_train, _ = complex_data_split
-    X_train["f1_categorical"] = X_train["f1_categorical"].astype("category")
+    return pd.DataFrame({"col_1": [1, 1, 1, 1], "col_2": [0, 0, 0, 0], "col_3": [1, 0, 1, 0]}, index=[1, 2, 3, 4])
 
-    return complex_lightgbm.fit(X_train, y_train)
+
+@pytest.fixture(scope="function")
+def y_train():
+    """
+    Fixture.
+    """
+    return pd.Series([1, 0, 1, 0], index=[1, 2, 3, 4])
```

Comment:
Maybe remove those """Fixture """ docstrings?

Reply:
removed in latest pr